donkeycar/parts/web_controller/templates/wsTest.html

```diff
@@ -0,0 +1,62 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8" name="viewport" content="width=device-width, initial-scale=1">
+    <title>Websocket Test</title>
+
+</head>
+    
+    
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.12.1/jquery.min.js"></script>
+        
+<script>
+    
+
+$(document).ready(function(){
+
+    
+    var socket = new WebSocket('ws://127.0.0.1:8887/wsDrive');
+
+    socket.onopen = async function(event){
+        console.log('Connected'); 
+        
+        console.log("Sending 1000 instructions\n")
+        while (true) {
+            data= {"angle": 1.0, "throttle": Math.random(), "recording": true, "drive_mode": "user"}
+            socket.send(JSON.stringify(data))
+        
+            await sleep(Math.random() * 10);
+        }
+    }
+
+    socket.onmessage = function(event){
+        
+        console.log(event.data);
+    };
+    
+    socket.addEventListener('error', function (event) {
+        console.log(event)
+    });
+    
+    
+    socket.addEventListener('close', (event) => {
+        console.log('The connection has been closed successfully.');
+    });
+
+
+    $(window).unload(function(event){
+        console.log('socket closed'); 
+        socket.close();
+    });
+    
+    
+    function sleep(ms) {
+        // console.log("sleeping " + ms + " ms")
+        return new Promise(resolve => setTimeout(resolve, ms));
```

Comment:
For completeness hold on to the `reject`or and `clearTimeout` when that is called.